hail/src/main/scala/is/hail/expr/types/TNDArray.scala

```diff
@@ -0,0 +1,36 @@
+package is.hail.expr.types
+
+import is.hail.annotations._
+import is.hail.expr.types.physical.PNDArray
+import org.apache.spark.sql.Row
+
+import scala.reflect.{ClassTag, _}
+
+final case class TNDArray(elementType: Type, override val required: Boolean = false) extends ComplexType {
+  val representation: Type = TStruct(
+    "flags" -> TInt64Required,           // encodes data layout
+    "shape" -> TArray(TInt64Required),   // length is ndim
+    "offset" ->  TInt64Required,         // offset into data
+    "strides" -> TArray(TInt64Required), // stride in each dim, length is 1
```

Comment:
Length is 1?  What's that mean?  Isn't the length ndim?

Reply:
yes, meant `ndim`, that was a typo, will fix in next PR